src/app/_components/chat/Chat.tsx

```diff
@@ -0,0 +1,258 @@
+  }, [isLoading, sendChatPreview]);
+
+  useEffect(() => {
+    if (isError) {
+      toast.error('채팅 불러오기에 실패 했습니다.');
+    }
+  }, [isError]);
+
+  const { ref: lastPrevChatRef } = useIntersect(
+    loadPrevChatHandler,
+    CHAT_INTERSECT_REF_OPTIONS,
+  );
+
+  const isDifferentDay = (
+    previousDateString: Date,
+    currentDateString: Date,
+  ) => {
+    const previousDate = new Date(previousDateString);
+    const currentDate = new Date(currentDateString);
+
+    const isDifferent =
+      previousDate.getDate() !== currentDate.getDate() ||
+      previousDate.getMonth() !== currentDate.getMonth() ||
+      previousDate.getFullYear() !== currentDate.getFullYear();
+
+    return isDifferent;
+  };
+
+  const cancelMessageHandler = () => {
+    if (confirm('해당 메시지 전송을 취소 하시겠습니까?')) {
+      cancelMessage();
+    }
+  };
+
+  const scrollDisabled = () => {
+    scrollRef.current = false;
+  };
+
+  return (
+    <div
+      ref={endOfMessagesRef}
+      className="h-0 w-full flex-grow overflow-auto overflow-x-hidden bg-gray-900 "
+    >
+      {isLoading && <ChatLoading count={12} />}
+      {isFetchingNextPage && scrollRef.current && (
+        <ChatLoading count={6} scrollDisabled={scrollDisabled} />
+      )}
+      {chats.map(({ id, content, createdAt, receiver, imageUrl }, index) => {
+        const isReceiver = !receiver[opponentId];
+        const beforeChat = chats[index - 1]?.createdAt;
+        const isFirstChat = isDifferentDay(beforeChat, createdAt);
+
+        return (
+          <Fragment key={id}>
+            {((!beforeChat && !hasNextPage) || (beforeChat && isFirstChat)) && (
+              <div className="my-3 flex w-full items-center gap-3 text-sm">
+                <hr className="flex-grow border-gray-500" />
+                {formatKoreanFullDate(createdAt)}
+                <hr className="flex-grow border-gray-500" />
+              </div>
+            )}
+            <div
+              className={`flex w-fit max-w-[84%] items-end gap-2 py-2 ${
+                isReceiver ? '' : 'ml-auto flex-row-reverse'
+              }`}
+              ref={
+                hasNextPage && index === 0
+                  ? lastPrevChatRef
+                  : index === chats.length - 1
+                  ? newChatRef
+                  : undefined
+              }
+            >
+              <div
+                className={`w-fit break-all rounded-t-lg ${
+                  imageUrl ? 'p-3' : 'px-4 py-2'
+                } ${
+                  isReceiver
+                    ? 'ml-2 rounded-r-lg bg-main-color-transparent'
+                    : 'mr-2 rounded-l-lg bg-white'
+                }`}
+              >
+                {content ? (
+                  content
+                ) : imageUrl ? (
+                  <Image
+                    width={261}
+                    height={167}
+                    src={imageUrl}
+                    alt="커넥션 채팅 이미지"
+                  />
+                ) : null}
```

Comment:
```suggestion
                { content && content }
                { imageUrl && 
                  <Image
                    width={261}
                    height={167}
                    src={imageUrl}
                    alt="커넥션 채팅 이미지"
                  />
             }
```
삼항 연산자 안에 삼항 연산자라 이게 더 좋을 듯 하네욤

Reply:
그러게 한번 더 봐볼껄 고마우이 💯 하기 코드로 변경 해볼게
```js
                {content ||
                  (imageUrl && (
                    <Image
                      width={261}
                      height={167}
                      src={imageUrl}
                      alt="커넥션 채팅 이미지"
                    />
                  ))}
```